core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -32,25 +35,35 @@ import org.apache.kafka.common.requests._
 import org.apache.kafka.common.requests.ApiVersionsResponse.ApiVersion
 import org.apache.kafka.common.requests.DescribeGroupsResponse.GroupMetadata
 import org.apache.kafka.common.requests.OffsetFetchResponse
-import org.apache.kafka.common.utils.{Time, Utils}
+import org.apache.kafka.common.utils.{KafkaThread, Time, Utils}
 import org.apache.kafka.common.{Cluster, Node, TopicPartition}
 
 import scala.collection.JavaConverters._
 import scala.util.Try
 
 class AdminClient(val time: Time,
                   val requestTimeoutMs: Int,
-                  val retryBackoffMs: Int,
+                  val retryBackoffMs: Long,
                   val client: ConsumerNetworkClient,
                   val bootstrapBrokers: List[Node]) extends Logging {
 
+  @volatile var running: Boolean = true
+
+  val networkThread = new KafkaThread("admin-client-network-thread", new Runnable {
+    override def run() {
+      while (running) {
+        client.poll(Long.MaxValue)
+      }
+    }
+  }, true)
```

Comment:
Yeah `NetworkClient` is not thread-safe. But `ConsumerNetworkClient` explicitly stated in its Java doc that it is thread-safe.